src/main/java/Parser.java

```
@@ -0,0 +1,106 @@
+        } else {
+            System.out.println("Please input a valid unmark");
+        }
+    }
+
+    private static void processMark(String command, TaskManager taskManager) {
+        String[] arrOfCommand = command.split(" ");
+        if (arrOfCommand.length > 1 && isInteger(arrOfCommand[1])) {
+            taskManager.markTasks(true, Integer.parseInt(arrOfCommand[1]));
+        } else {
+            System.out.println("Please input a valid mark");
+        }
+    }
+
+    private static void processNewEvent(String command, TaskManager taskManager) {
+        int spacePosition = command.indexOf(" ");
+        int dividerPosition = command.indexOf("/");
+        if(spacePosition < 0 || dividerPosition < 0 || dividerPosition + 2 > command.length()) {
+            System.out.println("Please input a valid event");
+            return;
+        }
+        String taskName = command.substring(spacePosition + 1, dividerPosition);
+        String at = command.substring(dividerPosition + 1);
+        taskManager.addNewEvent(taskName, at);
+    }
+
+    private static void processNewDeadline(String command, TaskManager taskManager) {
+        int spacePosition = command.indexOf(" ");
+        int dividerPosition = command.indexOf("/");
+        if(spacePosition < 0 || dividerPosition < 0 || dividerPosition + 2 > command.length()) {
+            System.out.println("Please input a valid deadline");
+            return;
+        }
+        String taskName = command.substring(spacePosition + 1, dividerPosition);
+        String by = command.substring(dividerPosition + 1);
+        taskManager.addNewDeadline(taskName, by);
+    }
+
+    private static void processNewTodo(String command, TaskManager taskManager) {
+        int spacePosition = command.indexOf(" ");
+        if(spacePosition < 0) {
+            System.out.println("Please input a valid todo");
+            return;
+        }
+        taskManager.addNewTodo(command.substring(spacePosition + 1));
+    }
```

Maybe you can define and let the function throw Custom Duke Exception to check "invalid task" instead of if/else in each of them